### ISSUE TITLE
Any chance of getting nscale working on google platform?

### ISSUE THREAD

**Issue by dreamineering** [nearform/nscale#10]
IDEA

With google you can get $500 credit for three months for startup ideas

https://developers.google.com/startups/

Was thinking it could be a sweet promotional idea to have 'startup kit' based on seneca and nscale to show how you can get started with micro-services for your MVP?

https://cloud.google.com/compute/docs/quickstart

This is why I am interested in nscale at least but not sure how to proceed, but will happily contribute what I can and help document the process

**Comment by mansona** [comment]
Has there been any progress on this?

**Comment by mcollina** [comment]
Not at all. Now you can use nscale with any system in which you can SSH into, _but_ you would still need to provision the machines yourself in Google platform.

However, adding support for it might not be too hard. We are planning to add support for a lot more platforms in the future.

**Comment by mansona** [comment]
thanks for the update @mcollina . I've taken a look at the AWS container repo https://github.com/nearform/aws-ami-container and it looks like there is the makings of an api in lib/container.js : 
```
{
    build: build,
    deploy: deploy,
    start: start,
    stop: stop,
    link: link,
    unlink: unlink,
    undeploy: undeploy,
    add: deploy,
    remove: undeploy
  }
```
I assume if one were to implement all of these commands specifically for the google platform then this process would be "done"? Is there any documentation as to what the expected behaviour of each of these commands is? maybe a test suite? :wink:

**Comment by mcollina** [comment]
It is not just that, also you would need the equivalent of https://github.com/nearform/nscale-aws-analyzer.

Unit-testing these things is almost impossible :(. Using mocks does not give any guarantee that the thing works, unfortunately.

The best shot of a test for nscale-aws-analyzer is @McDonnellDean's
https://github.com/McDonnellDean/nscale-aws-analyzer/tree/0c9d3a81aba5db7e716497c86929b06aab94db8e

A stub for a container module that you can fork from is https://github.com/nearform/blank-container/blob/master/lib/container.js. There all the API is documented.

**Comment by McDonnellDean** [comment]
@mansona Just to note, the link above is now in the main repo. I am currently working on expanding out the test suite for the AWS Analyzer but as Matt says it's not the easiest thing in the world to test. I'm looking at about 10 days between now and next PR with a full suite of tests.
